crates/forge/src/cheatcodes_hint_processor.rs

```diff
@@ -282,14 +282,19 @@ fn execute_syscall(
     let mut buffer = MemBuffer::new(vm, system_ptr);
 
     let selector = buffer.next_felt252().unwrap().to_bytes_be();
+
+    if std::str::from_utf8(&selector).unwrap() != "CallContract" {
+        return Err(HintError::CustomHint(Box::from(
+            "starknet syscalls cannot be used in tests".to_string()
+        )));
+    }
```

Comment:
I think you are right that it is the only case in which it happens, but let's ask @MaksymilianDemitraszek and @cptartur

Reply:
I think it's only syscalls, but not 100% sure about the internals of cairo code